backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -109,24 +107,16 @@ private async void OnMissionRunCreated(object? sender, MissionRunCreatedEventArg
 
         private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
         {
-            _logger.LogInformation(
-                "Triggered RobotAvailable event for robot ID: {RobotId}",
-                e.RobotId
-            );
-            var robot = await RobotService.ReadById(e.RobotId, readOnly: true);
-            if (robot == null)
+            if (e.Robot == null)
             {
-                _logger.LogError(
-                    "Robot with ID: {RobotId} was not found in the database",
-                    e.RobotId
-                );
+                _logger.LogError("OnRobotAvailable was triggered with null Robot value");
                 return;
             }
 
             _startMissionSemaphore.WaitOne();
             try
             {
-                await MissionScheduling.StartNextMissionRunIfSystemIsAvailable(robot);
+                await MissionScheduling.StartNextMissionRunIfSystemIsAvailable(e.Robot);
```

Comment:
rather than passing the entire robot object, would it make sense to only pass the relevant information about the robot needed to start the next mission

Reply:
This is out of scope for this PR